boa/src/builtins/string/mod.rs

```diff
@@ -266,6 +267,39 @@ impl String {
         }
     }
 
+    /// `String.prototype.at ( index )`
+    ///
+    /// This String object's at() method returns a String consisting of the single UTF-16 code unit located at the specified position.
+    /// Returns undefined if the given index cannot be found.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/proposal-relative-indexing-method/#sec-string-prototype-additions
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/at
+    pub(crate) fn at(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        let this = this.require_object_coercible(context)?;
+        let s = this.to_string(context)?;
+        let len = s.len();
+        let relative_index = args
+            .get(0)
+            .cloned()
+            .expect("No index provided")
```

Comment:
This will panic if no index is provided. The usual behaviour in the spec when this happens is to use the "undefined" value. I would therefore change it:

```suggestion
            .unwrap_or_default()
```